### DISCUSSION TITLE
How to resolve nodes addresses without BR

### DISCUSSION THREAD

**Discussion by nandra** [openthread/openthread#9853] [Q&A] [answered]
Hello, I'm experimenting with esp32-ha and using documentation I created simple thread network. I read documentation but I cannot find any mention about how to resolve nodes ip adresses or some kind of discovery. IS it possible without border router please? Thanks.

**Comment by jwhui** [comment, marked as answer]
DNS-based Service Discovery (DNS-SD) is generally used to discover services and their associated IP addresses. Any Thread device can offer a DNS/SRP Server and that device does not need to be a Thread Border Router.
